src/cow-react/api/uniswap/hooks.ts

```diff
@@ -0,0 +1,53 @@
+import { atom, useAtomValue, useSetAtom } from 'jotai'
+import useSWR from 'swr'
+import { getTokens } from './api'
+import { FetchTokensApiResult, FetchTokensResult } from './types'
+
+function isValidQuery(query: string): boolean {
+  return typeof query === 'string' && query.length > 0
+}
+
+const tokenLogoCacheAtom = atom<Map<number, Map<string, string>>>(new Map<number, Map<string, string>>())
```

Comment:
Nitpick/suggestion: create a type just to make the type more readable

Something like: `type TokenLogoCache = Map<number, Map<string, string>>`

Unless you think that's worse since the type name doesn't make it explicit it's a map of maps.
Not a strong opinion

Reply:
It is a good idea - especially in terms of highlighting what is a shared type, so to speak.

I've updated it - and added a small utility type for creating this in a more readable format.